### ISSUE TITLE
Unexpected (?) behaviour during eval mode

### ISSUE THREAD

**Issue by cemanil** [facebookresearch/higher#55]
Hi,

I've noticed that in the ```examples/maml-omniglot.py``` script, the model is kept at training mode during evaluation. That is, instead of calling ```model.eval()``` at the top of the evaluation function, ```model.train()``` is called. 

When I modify the script so that the model is in eval mode during evaluation, the validation accuracy drops to close to random guess. 

Do you mind explaining why this is happening? Having more insight into this behaviour can be important when using operations like dropout or batch norm in the network architecture. 

Thank you!

**Comment by ahaliassos** [comment]
I've also been having a similar problem. It seems that the running mean and variance of the original model's batch norm layers are not being updated when the forward function of the patched module is called. I would also really appreciate a response on this issue!

**Comment by egrefen** [comment]
This is @bamos’s code so maybe he can leave a quick comment? Otherwise I will look into it, but this will have to wait until after the NeurIPS deadline (unless it’s urgent?).

**Comment by bamos** [comment]
Hi, perhaps we should have added some more comments around this part of the code. Having the network in training model when it's being evaluated is intentional (and is what the original MAML code does), and lets us not have to worry about keeping track of the batch norm statistics, which is why it doesn't work in eval mode. Keeping track of the batch norm statistics when unrolling the model can be non-trivial, as discussed in the MAML++ paper, because the statistics may be different depending on what step of the unrolling you're at. It may be worth adding in warnings around this, and thinking about how we could add in ways of better-handling the different batch norm options so it doesn't have to manually be managed.

**Comment by cemanil** [comment]
Thank you for the quick response and explanation.
